pkg/kots/lint.go

```diff
@@ -344,6 +361,104 @@ func lintWithKubevalSchema(renderedFiles SpecFiles, originalFiles SpecFiles, sch
 	return lintExpressions, nil
 }
 
+func checkIfKotsVersionExists(version string) (bool, error) {
+	url := "http://api.github.com/repos/replicatedhq/kots/commits/%s"
+	token := os.Getenv("GITHUB_API_TOKEN")
+	var bearer = "Bearer " + token
+
+	if !strings.HasPrefix(version, "v") {
+		version = "v" + version
+	}
+
+	rwMutex.RLock()
+	verIsCached := kotsVersions[version]
+	rwMutex.RUnlock()
+
+	if !verIsCached {
+		req, err := http.NewRequest("GET", fmt.Sprintf(url, version), nil)
+		if err != nil {
+			return false, errors.Wrap(err, "failed to create new request")
+		}
+		req.Header.Set("Authorization", bearer)
+		client := &http.Client{}
+		resp, err := client.Do(req)
+		if resp.StatusCode == 422 {
+			return false, nil
+		} else if resp.StatusCode == 200 {
+			rwMutex.Lock()
+			kotsVersions[version] = true
+			rwMutex.Unlock()
+		} else {
+			return false, errors.New(fmt.Sprintf("received non 200 status code (%d) from GitHub API request", resp.StatusCode))
```

Comment:
this doesn't log the actual error that was returned by github. you can get the body of the response with something like this
```go
b, err := ioutil.ReadAll(resp.Body)
...
```